tokio/src/runtime/task/trace/mod.rs

```diff
@@ -0,0 +1,215 @@
+use backtrace::BacktraceFrame;
+use std::future::Future;
+use std::{
+    cell::Cell,
+    ffi::c_void,
+    fmt,
+    pin::Pin,
+    ptr::{self, NonNull},
+    task::{self, Poll},
+};
+
+mod symbol;
+mod tree;
+
+use symbol::Symbol;
+use tree::Tree;
+
+type Backtrace = Vec<BacktraceFrame>;
+type SymbolTrace = Vec<Symbol>;
+
+/// The ambiant backtracing context.
+pub(crate) struct Context {
+    /// The address of [`Trace::root`] establishes an upper unwinding bound on
+    /// the backtraces in `Trace`.
+    active_frame: Cell<Option<NonNull<Frame>>>,
+    /// The place to stash backtraces.
+    collector: Cell<Option<Trace>>,
+}
+
+/// A [`Frame`] in an intrusive, doubly-linked tree of [`Frame`]s.
+struct Frame {
+    /// The location associated with this frame.
+    inner_addr: *const c_void,
+
+    /// The parent frame, if any.
+    parent: Option<NonNull<Frame>>,
+}
```

Comment:
What is the purpose of this linked tree? Is there any reason we can't just use a `Vec<*const c_void>` in the context to store this information?